test/es6/rlexe.xml

```diff
@@ -868,6 +868,13 @@
       <tags>exclude_arm</tags>
     </default>
   </test>
+  <test>
+    <default>
+      <files>generators-functionality.js</files>
+      <compile-flags>-ES6Generators -ES6Classes -ES6DefaultArgs -debuglaunch -args summary -endargs</compile-flags>
```

Comment:
Note that a lot of existing generators-functionality.js tests were already failing/crashing if we added the -debuglaunch flag.

Reply:
interesting.

---
In reply to: [201881156](https://github.com/Microsoft/ChakraCore/pull/5451#discussion_r201881156) [](ancestors = 201881156)

Reply:
Could you  add with async functionality as well?

---
In reply to: [201881720](https://github.com/Microsoft/ChakraCore/pull/5451#discussion_r201881720) [](ancestors = 201881720,201881156)

Reply:
Do we already have test case which has the failing scenario in the bug? If not can you please add it?

Reply:
Can you please add a test case with the same scenario where we catch first chance exception? Also a test case with finally block?

Reply:
There is already a test case with the same scenario: 
```
        name: "Test cases with return api invoked while the generator resumes inside one of the try-catch-finally block",
        body: function () {
            var gf = function* () { try { yield 1; } catch (ex) { }}
            var g = gf();
            assert.areEqual({ value: 1, done: false }, g.next(), "First next call should return 1 from try block");
            assert.areEqual({ value: 2, done: true }, g.return(2), "Return statement completes the function from try block");
            assert.areEqual({ value: undefined, done: true }, g.next(), "Generator is already in completed state");
```

Reply:
There is also an existing test with same scenario + finally block:

```
            gf = function* () { try { throw 1; } catch (e) { yield e; } finally { yield 2; return 3; } }
            g = gf();
            assert.areEqual({ value: 1, done: false }, g.next(), "First next call should return 1 from the catch block");
            assert.areEqual({ value: 2, done: false }, g.return(100), "Return causes the finally block to execute");
            assert.areEqual({ value: 3, done: true }, g.next(), "Second next call should return 3 from the end of finally block");
            assert.areEqual({ value: undefined, done: true }, g.next(), "Method execution has finished");
```